lesson16/main.js

```diff
@@ -0,0 +1,125 @@
+      if(responseData.length === 0) {
+         displayMessage("データはまだありません");
+         return;
+         } 
+   } catch (error) {
+      displayMessage(error);
+      console.log(error);
+   } finally {
+      endLoading();
+   }
+   createCategoryTab(responseData);
+   renderTabContainer(responseData);
+}
+
+const createCategoryTab = (values) => {
+   const fragment = document.createDocumentFragment();
+   for (let i = 0; i < values.length; i++) {
+      const categoryTabItem = createElementWithClassName("li", "category-tab__item")
+      categoryTabItem.id = `js-categoryTabItem${i}`;
+      const categoryTabText = createElementWithClassName("a", "category-tab__text");
+      categoryTabText.id = `js-categoryTabText${i}`
+      categoryTabText.textContent = values[i].category
+      fragment.appendChild(categoryTabItem).appendChild(categoryTabText);
+   }
+   categoryTab.appendChild(fragment);
+   //初期表示のタブの選択はデータとして持っているがまだ理解できていないので、現段階ではNewsのタブとしてある
+   const newsTabItem = document.getElementById("js-categoryTabItem0");
+   newsTabItem.classList.add("is-active");
+}
+
+const renderTabContainer = (values) => {
+   const tabFragment = document.createDocumentFragment();
+   const tabContainer = createElementWithClassName("div", "tab-container");
+   tabContainer.id = "js-tabContainer";
+   const articlesData = values.map((value) => value.articles);
+   for (let i = 0; i < values.length; i++) {
+      const tabContents = createElementWithClassName("div", "tab-contents");
+      tabContents.id = `js-tabContents${i}`;
+      const articlesTitles = articlesData[i].map((value) => value.title);
+      const articlesHref = articlesData[i].map((value) => value.href);
+      const tabLists = createElementWithClassName("ul", "tab-contents__lists");
+      tabLists.id = `js-tabLists${i}`;
+      const tabImage = createElementWithClassName("img", "tab-contents__image");
+      tabImage.id = `js-tabImage${i}`;
+      tabImage.src = values[i].img;
+      console.log(values);
+      console.log(articlesTitles);
+      tabContents.appendChild(tabImage);
+      for (let j = 0; j < articlesTitles.length; j++) {
+         const tabItems = createElementWithClassName("li", "lists__item");
+         const tabItemsHref = createElementWithClassName("a", "lists__item__text");
+         tabItemsHref.href = articlesHref[j];
+         tabItemsHref.textContent = articlesTitles[j];
+         tabFragment.appendChild(tabContainer).appendChild(tabContents).appendChild(tabLists).appendChild(tabItems).appendChild(tabItemsHref);
+      }
+   }
+   body.appendChild(tabFragment);
+   //TODO:どのカテゴリタブを初期表示時に選んでいるかはデータとして持っている
+   //一旦、Newカテゴリーを初期表示としている
+   const newsContents = document.getElementById("js-tabContents0");
+   newsContents.classList.add("is-show");
+}
+
+categoryTab.addEventListener("click", (e) => {
+   const activeTab = document.getElementsByClassName("is-active")[0];
+   const showContents = document.getElementsByClassName("is-show")[0];
```

Comment:
nits
If you want to get the first element that matches the class name, you can also use document.querySelector.
```suggestion
  const activeTab = document.querySelector(".is-active");
  const showContents = document.querySelector(".is-show");
```

Reply:
Thank you for your advise.
I didn't know this way.
I changed that way to write. 
Please confirm that part.
[3c2f125](https://github.com/taka0209/js_lesson/pull/18/commits/3c2f125ee93815acd9252a8b05f1172ac58d3d90)